### ISSUE TITLE
gpclient not reading /etc/gpservice/gp.conf?

### ISSUE THREAD

**Issue by altosys** [yuezk/GlobalProtect-openconnect#188]
I'm having issues connecting to a VPN gateway using gpclient, it seems to me that it's not loading config from `/etc/gpservice/gp.conf` properly, as changes made in that file don't really seem to affect anything.

For instance, setting `--clientos=Mac` in the config file still results in clientos=Linux being sent in the login request to the gateway.
I'm also specifying a cert file in the configuration, but the client keeps failing to login and reverts to asking for username and password, which don't exist.

My config:

```
[*]
openconnect-args=--certificate=/home/user/Downloads/cert.crt
        --sslkey=/home/user/Downloads/cert.key
        --protocol=gp
        --gateway
        --clientos=Mac
```

Using `gp-saml-gui` I can get this to work with the command below to get the SAML response and then successfully connect by running the openconnect command suggested by the output.

```
❯ gp-saml-gui portal.connect.mydomain.com --cert=/home/user/Downloads/cert.crt --key /home/user/Downloads/cert.key --gateway --clientos=Mac
```

I also checked with `strace` and I can't see that gpclient reads the `/etc/gpservice/gp.conf` file. How can I verify that the configuration is actually being read and used?

**Comment by yuezk** [comment]
Sorry about that, but this client doesn't support certificate authentication.

**Comment by dsa146** [comment]
The gp.conf seems not working for me as well. I am on Ubuntu 20.04 LTS.
I added a "--servercert" to the openconnect-args but it still does not allow the ca cert to be verified during the login process.

**Comment by D1Ck3n** [comment]
> Sorry about that, but this client doesn't support certificate authentication.

not at all? so i could not use my certificate.pfx to connect to my GP gateway? :/

**Comment by yuezk** [comment]
@D1Ck3n @dsa146 @altosys My VPN server doesn't eanble certificate authentication. So I haven't tested this case. But from the code perspective, it is not supported.

If your VPN server doesn't use the Web-based authentication (e.g., SAML/OIDC), you don't need this client. You could try https://gitlab.gnome.org/GNOME/NetworkManager-openconnect. Which is the official extension for the Network Manager and it integrates better with the system Network configuration.

**Comment by gowtham-acceldata** [comment]
@yuezk @dsa146 

I'm also facing this issue. I don't think the configuration file `/etc/gpservice/gp.conf` is being read/used by the gpclient. 
I used to use the settings input in the GUI window to pass custom parameters earlier, since the setting input is disabled and it shows the configurations has been moved to `/etc/gpservice/gp.conf` I tried adding the custom parameters to this file but I can see it's not being read by the gpclient. 

![image](https://user-images.githubusercontent.com/117887410/203766362-fdc1d9e0-4d2b-4257-afff-762bec5d2509.png)

^ My GlobalProtect Server uses a self-signed certificate and the certificate authority is invalid, I used to by-pass this issue by passing the argument `--servercert pin-sha256:xxxxxxxxxxxxxxxx` to the openconnect-args. Now I'm unable pass the parameters as the file `/etc/gpservice/gp.conf` is not being read by the gpclient.

I think this is also related to the below open issues: 
  * https://github.com/yuezk/GlobalProtect-openconnect/issues/33
  * https://github.com/yuezk/GlobalProtect-openconnect/issues/119
